pkg/virt-api/webhooks/validating-webhook/admitters/instancetype-admitter.go

```diff
@@ -117,17 +128,73 @@ func validateRequestResource(ar *admissionv1.AdmissionReview, resourceType strin
 	return nil
 }
 
-func validateDedicatedCPUPlacement(instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) []metav1.StatusCause {
-	if instancetypeSpec == nil {
-		return nil
+func validateInstanceTypeCPU(instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) []metav1.StatusCause {
+	if instancetypeSpec != nil {
+		hasRequestedCPU := false
+		if instancetypeSpec.Resources != nil {
+			_, hasRequestedCPU = instancetypeSpec.Resources.Requests[k8sv1.ResourceCPU]
+		}
+
+		if instancetypeSpec.CPU == nil {
+			if !hasRequestedCPU {
+				return []metav1.StatusCause{{
+					Type:    metav1.CauseTypeFieldValueRequired,
+					Message: "at least one of spec.resources.requests.CPU or spec.CPU should be defined",
+					Field:   k8sfield.NewPath("spec", "instancetype").String(),
+				}}
+			}
+			return nil
+		}
+
+		if hasRequestedCPU && instancetypeSpec.CPU.Guest > 0 {
+			return []metav1.StatusCause{{
+				Type:    metav1.CauseTypeFieldValueDuplicate,
+				Message: "spec.resources.requests.CPU and spec.CPU can not be both defined",
+				Field:   k8sfield.NewPath("spec", "instancetype").String(),
+			}}
+		} else if !hasRequestedCPU && instancetypeSpec.CPU.Guest == 0 {
+			return []metav1.StatusCause{{
+				Type:    metav1.CauseTypeFieldValueRequired,
+				Message: "at least one of spec.resources.requests.CPU or spec.CPU should be defined",
+				Field:   k8sfield.NewPath("spec", "instancetype").String(),
+			}}
+		}
 	}
 
-	if instancetypeSpec.CPU.DedicatedCPUPlacement {
-		return []metav1.StatusCause{{
-			Type:    metav1.CauseTypeFieldValueInvalid,
-			Message: "dedicatedCPUPlacement is not currently supported",
-			Field:   k8sfield.NewPath("spec", "cpu", "dedictatedCPUPlacement").String(),
-		}}
```

Comment:
Can you remove this in a standalone commit and add some functional test coverage?